cps/tasks/download.py

```diff
@@ -97,7 +97,14 @@ def run(self, worker_thread):
                         self.message = f"Successfully downloaded {self.media_url_link} to <br><br>{file_downloaded}"
                         new_video_path = response.json()["new_book_path"]
                         new_video_path = next((os.path.join(new_video_path, file) for file in os.listdir(new_video_path) if file.endswith((".webm", ".mp4"))), None)
-                        conn.execute("REPLACE INTO media (webpath, path) VALUES (?, ?)", (self.media_url, new_video_path))
+                        try:
+                            conn.execute("UPDATE media SET path = ? WHERE webpath = ?", (new_video_path, self.media_url))
+                            conn.commit()
+                        except sqlite3.IntegrityError as e:
+                            if "UNIQUE constraint failed" in str(e):
+                                pass
```

Comment:
this should probably be removed before merging

Reply:
The UPDATE operation will always encounter a "UNIQUE constraint failed" error upon every download.

Reply:
> xklb will not attempt to download something if it has been downloaded before.

Thanks to @chapmanjacobd who summarized the "xklb ethos" above! 💯

> “They laugh at me because I'm different; I laugh at them because they're all the same.”

Thanks also to Kurt Cobain for helping us all think about... _who among us is a mere clone (same primary key) and who is not!_ 😆

Now to dig a little deeper, in the spirit of AI² (Articulate Assumptions, Intentions, Intuition) here are a few questions to help @deldesir get on track and complete this PR:

1) Is @deldesir needing to partially bypass the xklb mechanism uses to define (what it views as!) duplicate[*] videos to move forward here?
2) Does xklb's mechanism presumably involve yt-dlp's "video download archive" feature, or some URL-as-master-key database checker or such?  (Assuming that videos at a particular URL are considered to be "kinda immutable" as a simplifying assumption built into xklb!)
3) Long term Questions... also matter this year (so I've thrown in a few notes and a screenshot below, to force us to start teasing out Purposeful Design Assumptions, needing to become more prominent very soon, even if we can't yet solve these today!!)
4) _Short Term: What's likely the most **pragmatic / solid option** to pull this PR together as "Good Enough", allowing us all the build upon later ?_  (Or, is there perhaps a completely different way of attacking this PR?!)

[*] THE FESTERING QUESTION FOR WEEKS AHEAD — _NOT TODAY!_ — AS WE DON'T YET HAVE A CLEAN ENOUGH COMMUNITY CONTEXT...

- e.g. "BUT I'VE ALREADY SEEN THAT MOVIE!"  (PEOPLE HAVE _SHARPLY_ DIFFERENT VIEWS OF WHAT "DUPLICATE VIDEO" MEANS!)  The [tension here](https://github.com/iiab/calibre-web/wiki#todo-supporting-redownloading-videos-with-the-same-id) is very real indeed.  And is likely related to the fact that Archivists, Historians, Journalists, Human Rights professionals, Legal folk, Educators, serious Students etc each have completely different opinions — based on very strong feelings in their heart-of-hearts  — when they each talk about what constitutes a "duplicate video..."

- e.g. if a YouTube video is trimmed, has important faces blurred, etc, it might still retain the very same URL and 11-character YouTube video ID.  But... (leaving all politics aside — i.e. regardless whether we call that a slippery slope to Orwellian censorship, or common-sense removal of child pornography scenes no longer tolerated in a civilized world)  certainly the video's no longer the same from an archivist's perspective.

- e.g. screenshot below from **"Can you change YouTube videos after uploading them?"** helps shine a bit of light on this murky (no doubt evolving) reality — at least hinting at how YouTube defined "duplicate videos" as of 5 years ago:

![image](https://github.com/iiab/calibre-web/assets/2458907/09fa9d44-8fac-4c98-9002-645653638e3e)

https://www.quora.com/Can-you-change-YouTube-videos-after-uploading-them

Reply:
I tried to work around this, without success, by using --download-archive flag (seems not functional). 

Looks like, when a playlist URL is submitted, video IDs are checked for existence per https://github.com/chapmanjacobd/library/blob/1f363ccd6b30ec11cd6eae3fbecd8e6f4f20a0b6/xklb/tube_backend.py#L121-L126. I guess the same mechanism is used for individual videos.